### ISSUE TITLE
Bloodborne graphics issues on Intel GPU

### ISSUE THREAD

**Issue by hgh32** [shadps4-emu/shadPS4#1739]
Long time ago I reported the assertion issues with Bloodborne on Intel GPU #765

There are two ways to skip the assertion
https://github.com/shadps4-emu/shadPS4/blob/main/src/video_core/renderer_vulkan/vk_rasterizer.cpp

Comment assertion code
---
`ASSERT(adjust % fmt_stride == 0);`

https://github.com/user-attachments/assets/5b1298ff-afe4-4025-a98b-a37c7e111c06

This will cause severe graphics issues
[shad_log.txt](https://github.com/user-attachments/files/18096975/shad_log.txt)

Use buffer offsets directly for alignment
---
change this line `const u32 offset_aligned = Common::AlignDown(offset, alignment);` to `const u32 offset_aligned = offset;`

https://github.com/user-attachments/assets/18d04bba-1abb-4e9a-8e54-48ddde618c4b

Seems just partially alleviate
[shad_log.txt](https://github.com/user-attachments/files/18096990/shad_log.txt)

---

I think that related to `0xfefebf9f` shader
When skipping it doesn't cause the assertion and the graphics issues are gone

**Update: This still has the display issues of UI text at 0:06**

https://github.com/user-attachments/assets/ac2f255e-92e9-4d6c-bbcd-f81c9f2a6b9b

Besides adding skip shader code, I didn't change anything else
[shad_log.txt](https://github.com/user-attachments/files/18096999/shad_log.txt)

# 
The last video is from  #1683 pr builds, as this fixes the assertion

https://github.com/user-attachments/assets/76e1e8bf-97c4-449a-928a-24f19cb36514

Still have graphics issues, flickering becomes very random and occurs again densely over time or in certain area
[shad_log.txt](https://github.com/user-attachments/files/18097076/shad_log.txt)

You can see they all have similar issues, screen flickering, vertex explosions and UI display issues

All tests with vanilla game without any mods and official builds

**Comment by carzhao** [comment]
i have the same issues (CPU- intel ultra7 258v)

**Comment by Missake212** [comment]
Hi, could you confirm this still happens on latest main?

**Comment by hgh32** [comment]
Yes, seems still happen on latest main

**Comment by Xphalnos** [comment]
Can you try again with more recent drivers and let us know if it is still relevant?

**Comment by hgh32** [comment]
> Can you try again with more recent drivers and let us know if it is still relevant?

I'm honestly not sure what exactly is causing this
But I really don't think updating the driver will work
Intel sets all types of buffer alignment to 64
Linux systems don't have these graphics issues because it uses open source graphics drivers

**Comment by Missake212** [comment]
I can confirm that on Windows with the latest drivers, Intel iGPUs still face issues in Bloodborne (don't take the horrible performance into account that's to be expected, I'm talking about the screen turning black every few seconds and the flashing multi colors light)
Here's a video showcasing the issue: 

https://youtu.be/YJZ2gHfh_AY

And here's a screenshot of my Intel app to show that my drivers are up to date:

![Image](https://github.com/user-attachments/assets/3c57aaf6-86d2-4136-83a2-8c52981a0fe1)

And here's my log to show that I'm not using any patches, keep in mind that for this test I had to disable my sfx folder inside my game files because otherwise I would crash, if needed I can provide another short video this time with the sfx folder activated to show that the issue still happens, also I'm running the game with absolutely no mods, it is a fresh install.

[shad_log.txt](https://github.com/user-attachments/files/18826498/shad_log.txt)

**Comment by guliguocmyk** [comment]
so is there any action that can be taken to resolve this issue?

**Comment by Missake212** [comment]
Unfortunately there's no fix atm. Pretty hard to debug as not many people have an Intel GPU.

**Comment by guliguocmyk** [comment]
Is there anything I can do to help this?

Missake212 ***@***.***> 于2025年2月27日周四 23:42写道：

**Comment by Missake212** [comment]
@guliguocmyk If you know how to use something like Renderdoc and can figure out why it's like that on Intel GPUs only then you can try, but as of right now I don't think there's much we can do beside waiting for someone with more knowledge to try to look into this problem.

**Comment by lmutig** [comment]
The Intel Arc A770, A750, and other graphics cards. After the driver update on February 27th, there don't seem to be any issues anymore.

**Comment by hgh32** [comment]
It seems that Intel has finally fixed it.
The flickering issue no longer occurs with the latest graphics driver (32.0.101.6632).
@Missake212 Can you retest it again and I will close this issue if you confirm

**Comment by hgh32** [comment]
> I can confirm that on Windows with the latest drivers, Intel iGPUs still face issues in Bloodborne (don't take the horrible performance into account that's to be expected, I'm talking about the screen turning black every few seconds and the flashing multi colors light) Here's a video showcasing the issue:
> 
> https://youtu.be/YJZ2gHfh_AY
> 
> And here's a screenshot of my Intel app to show that my drivers are up to date:
> 
> ![Image](https://github.com/user-attachments/assets/3c57aaf6-86d2-4136-83a2-8c52981a0fe1)
> 
> And here's my log to show that I'm not using any patches, keep in mind that for this test I had to disable my sfx folder inside my game files because otherwise I would crash, if needed I can provide another short video this time with the sfx folder activated to show that the issue still happens, also I'm running the game with absolutely no mods, it is a fresh install.
> 
> [shad_log.txt](https://github.com/user-attachments/files/18826498/shad_log.txt)

You seem to be using a very old driver
There is a version bug in the Intel control panel
You should download the latest driver from their official website

**Comment by Missake212** [comment]
You're right, it was the "WHQL Certified Driver Updates" option that didn't let me get the latest driver, I tested again and no issues on my Intel iGPU either, you can close it, that's nice to hear !
